data/cassandra.go

```diff
@@ -125,42 +126,65 @@ in the PRIMARY KEY).
 
 Note that Delete will still be able to execute the DELETE query if "id" is specified as a primary_key_column.
 */
-func (c CassandraConnector) Delete(table, id string, primary_key_cols ...string) error {
-	query := "DELETE FROM " + table + " WHERE id = " + id // Base query
-
-	// If the PRIMARY KEY is compound (uses more columns than "id"), then append those onto the base query
-	for _, col := range primary_key_cols {
-		if col == "id" { // Skip the "id" column; it's already in the base query
+func (c CassandraConnector) Delete(table, id string, opts CassandraQueryOptions) error {
+	query := "DELETE FROM " + table + " WHERE " // Base query
+	c.Infof("Deleting:%s,%s,%v", table, id, opts)
+	firstField := true
+	if id == "" && len(opts.Filters) == 0 {
```

Comment:
This makes me a little nervous without testing.  Let's work together on how to refactor this with testing and share the logic with the FilteredReadAll